common/os.go

```diff
@@ -148,6 +148,9 @@ func WriteFileAtomic(filename string, data []byte, perm os.FileMode) error {
 	} else if n < len(data) {
 		return io.ErrShortWrite
 	}
+	// Close the file before renaming it, otherwise it will cause "The process 
+	// cannot access the file because it is being used by another process." on windows or 
+	// cause "cross-link error" on linux when you try to save it to another partition.
```

Comment:
note we're using the same dir where filename is located, so cross-link error can't happen

Reply:
yes.